core/sql/parser/StmtDDLCreate.cpp

```diff
@@ -5007,7 +5020,31 @@ StmtDDLCreateTable_visitTableDefElement(ElemDDLNode * pCreateTableNode,
   StmtDDLCreateTable * pCreateTable =
     pCreateTableNode->castToStmtDDLCreateTable();
 
-  if (pElement->castToElemDDLConstraint() NEQ NULL)
+  if (pElement->castToElemDDLLikeCreateTable() NEQ NULL)
+    {
+      pCreateTable->likeSourceTableCorrName_ = 
+        pElement->castToElemDDLLikeCreateTable()
+        ->getDDLLikeNameAsCorrName();
+      pCreateTable->likeOptions_ = 
+        pElement->castToElemDDLLikeCreateTable()
+        ->getLikeOptions();
+
+      if ((NOT pCreateTable->isExternal()) &&
+          (pElement->castToElemDDLLikeCreateTable()->forExtTable()))
+        {
+          *SqlParser_Diags << DgSqlCode(-3242)
+                           << DgString0("'for' clause can only be specified when creating an 'external' table.");
```

Comment:
It looks like the style of SqlciErrors.txt is to put syntax elements in upper case in error messages, unquoted. So, for example, "FOR clause can only be specified when creating an external table."